src/libtcod/zip_c.c

```diff
@@ -362,6 +368,15 @@ TCOD_console_t TCOD_zip_get_console(TCOD_zip_t pzip) {
 	return ret;
 }
 
+TCODLIB_API TCOD_random_t TCOD_zip_get_random(TCOD_zip_t zip)
+{
+	TCOD_random_t ret;
+	size_t s = TCOD_zip_get_int(zip);
+	ret = (TCOD_random_t)malloc(s);
+	TCOD_zip_get_data(zip, s, ret);
+	return ret;
+}
```

Comment:
This doesn't look like it'd be forward compatible with changes to `TCOD_Random`.  If the struct was ever extended then the extended part would not be in allocated memory.  This also wouldn't support other kinds of changes to the struct.